### PR TITLE
Production dependencies

### PR DESCRIPTION
**Pull request by bkuhl**
https://github.com/Realpage/laravel/pull/7 should go first.

 * Builds route cache for `staging` and `master` branches
 * Removes dev dependencies

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,19 +31,30 @@ before_install:
...
+  # Seed database
   - docker exec -it $(docker-compose ps -q fpm) php artisan migrate --seed
+  # Run tests
+  - docker exec -it $(docker-compose ps -q fpm) vendor/bin/phpunit
+  # Install npm dependencies & run gulp
+  - docker run -i -v $(pwd):/var/www/html $(docker images -q realpage/builder:npm-3) npm rebuild node-sass && npm install --silent
+  - docker run -i -v $(pwd):/var/www/html $(docker images -q realpage/builder:npm-3) npm install --global gulp && gulp --production
 after_success:
+  # Switch to production-only dependencies
+  - if [[ $TRAVIS_BRANCH == 'master' || $TRAVIS_BRANCH == 'staging' ]]; then docker exec -it $(docker-compose ps -q fpm) rm -rf /var/www/html/vendor ; fi
```
Comment (bkuhl) [inline .travis.yml]: I'm curious what your thoughts are on this... a bad container shouldn't be deployed if the health checks pass.

The more I think about this... maybe we should install phpunit/phpcs separately and not use what's in the composer file so we're not required to install dev dependencies...

**Comment by bkuhl** [comment]
@millidavids @jlaswell

**Comment by bkuhl** [comment]
:pager:

**Comment by millidavids** [comment]
LGTM, looks like you may need to rerun on Travis though 😭

**Comment by jlaswell** [comment]
That will fail again, because route caching only works for controller routes. Callbacks can't be cached and will cause a failure.

**Comment by bkuhl** [comment]
It passed ^_^

**Comment by bkuhl** [comment]
:+1:

**Pull request merged**